lib/debt_management_center/sharepoint/errors.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module DebtManagementCenter
+  module Sharepoint
+    class Errors < Faraday::Middleware
+      def on_complete(env)
+        return if env.success?
+
+        Rails.logger.error("Sharepoint failed path: #{env.url.path[%r{.*/Web/[^/]+}]}")
```

Comment:
Will you have a Datadog monitor looking for this error (and the one in `pdf_errors.rb`)? Or will you just monitor Datadog as needed?

If you have the ability to and want to make a Log Monitor in datadog, it could be useful to prefix both of these errors with something that is identifiable so the log monitor could look for either errors. Something like, "Debt Management Center: Sharepoint failed path..." and "Debt Management Center: Sharepoint PDF failed..." for the other one.

Reply:
Hey @rmtolmach,

Thank you for your review.

> Will you have a Datadog monitor looking for this error (and the one in pdf_errors.rb)? Or will you just monitor Datadog as needed?

We have a monitor that catches errors associated with this request and others. You can find it here: `modules/debts_api/app/models/debts_api/v0/form5655_submission.rb#set_vha_completed_state`. It's reporting and we are storing error messages in the DB too. I think we're covered but I'm happy to add a monitor for `errors.rb` and `pdf_errors.rb` specifically :) if you'd like. WDYT?

> ... it could be useful to prefix both of these errors with something that is identifiable so the log monitor could look for either errors. Something like...

Nice idea, I've added it 👍🏻